### DISCUSSION TITLE
How to enhanced transformation (e.g. for currency number values)

### DISCUSSION THREAD

**Discussion by maku** [pablo-abc/felte#125] [Q&A]
I would like to have extensive transformation capabilities. e.g. for number fields like currencies etc.

Imagine the use case:

im my object I have a field (say amount) which holds a currency value as "number".
e.g. 1299.99

The presentation in the input should be "1.299,99 €" (therefore a formatting is needed  because of localization needs). 
When the input field gets the focus the presentation should change to "1299,99" 
When the input lose the focus the presentation should be "1.299,99 €"
In the data object I need the number value.

How can I achieve this in felte?

TIA

**Comment by pablo-abc** [comment]
Sorry about the delay here. Currently there is no way to handle "masking" directly from Felte. There's an open issue I created for it in #82 but I haven't had the time to work on it (since it's not something I've actually needed to use yet).

I imagine using a general masking library + a transform function to transform the string to a number should work, though. If this does not work somehow feel free to open an issue (or add to #82)
